specs/Standards/NonFungibleToken/Payout.md

```diff
@@ -80,18 +83,27 @@ pub trait Payouts{
     receiver_id: AccountId,
     token_id: String,
     approval_id: u64,
+    memo: Option<String>,
     balance: U128,
-    max_len_payout: u32,
-  ) -> Payout{
+    max_len_payout: Option<u32>,
+  ) -> Payout {
     assert_one_yocto();
     let payout = self.nft_payout(token_id, balance);
-    self.nft_transfer(receiver_id, token_id, approval_id);
+    self.nft_transfer(receiver_id, token_id, approval_id, memo);
     payout
   }
 }
 ```
 
-Note that NFT and financial contracts will vary in implementation. This means that some extra CPU cycles may occur in one NFT contract and not another. Furthermore, a financial contract may accept fungible tokens, native NEAR, or another entity as payment. Transferring native NEAR tokens is less expensive in gas than sending fungible tokens. For these reasons, the maximum length of payouts may vary according to the customization of the smart contracts.
+## Fallback on error
+
+In the case where either the `max_len_payout` causes a panic, or a malformed `Payout` is returned, the caller contract should transfer all funds to the token owner.
```

Comment:
Example scenario background information:
- Benji puts token A for sale for 100 NEAR. 
- token A has a payout object which dictates that Mike should get 20 NEAR and Benji should get 80 NEAR. 
- Willem comes along and purchases the NFT for 100 NEAR on the marketplace.

Logic: 
- Marketplace contract calls `nft_transfer_payout` on the NFT contract which will transfer the NFT to willem and send back a payout object. 
- That payout object as per the current standard was wrapped in a struct. Your new changes propose it should just be a type (just a hashmap). 

In your new proposal: 
- If the NFT contract sends back a struct: 
```
{
  "payout": {
    "mike": "20000000000000000000000000",
    "benji": "80000000000000000000000000",
  }
}
```
This should transfer all the funds (100 NEAR) to the token owner, which is Willem in this case. That means Willem got the NFT for free essentially (which is fine because so sorry Benji, but your NFT was minted on an improper contract that didn't implement the standards correctly). 

- If the NFT contract sends back a payout object: 
```
{
    "mike": "20000000000000000000000000",
    "benji": "80000000000000000000000000",
}
```
Everything would be fine and the marketplace should pay mike 20 NEAR and benji 80 NEAR.

Reply:
I'll change it to original `token_owner`.  The seller should be the default.